Docs: 10.4 technical note for alertingUpgradeDryrunOnStart (#83262)
grafana/grafana: docs/sources/upgrade-guide/upgrade-v10.4/index.md

{{< docs/shared lookup="upgrade/upgrade-common-tasks.md" source="grafana" version="<GRAFANA VERSION>" >}}

## Technical notes

> ### Legacy alerting -> unified alerting dry-run on start

Reviewer: we should call this Grafana Alerting and not unified alerting

> In preparation for the removal of Legacy alerting in v11, Grafana will initiate a dry-run of the upgrade from legacy alerting to unified alerting on every startup if an instance is not already upgraded. This operation will log the outcomes of the upgrade attempt and identify any issues requiring attention before the actual upgrade can be successfully executed. No changes will be made during the dry-run.

Reviewer: ```suggestion
If you haven't already upgraded to Grafana Alerting from legacy Alerting, Grafana will initiate a dry-run of the upgrade every time you log into your instance. This is in preparation for the removal of legacy Alerting in Grafana v11. The dry-run logs the results of the upgrade attempt and identifies any issues requiring attention before you can successfully execute the upgrade.. No changes are made during the dry-run.
```

> This behaviour can be disabled via the feature flag `alertingUpgradeDryrunOnStart`:

Reviewer 2: ```suggestion
You can disable this behaviour using the feature flag `alertingUpgradeDryrunOnStart`:
```

```toml
[feature_toggles]
alertingUpgradeDryrunOnStart=false
```

{{% admonition type="note" %}}

> Users are strongly encouraged to review the upgrade guide available at https://grafana.com/docs/grafana/v10.4/alerting/set-up/migrating-alerts/ and perform the necessary upgrade steps prior to v11.

Reviewer 2: ```suggestion
We strongly encourage you to review the [upgrade guide](https://grafana.com/docs/grafana/v10.4/alerting/set-up/migrating-alerts/) and perform the necessary upgrade steps prior to v11.
```
